pyccel/ast/core.py

```diff
@@ -730,15 +730,12 @@ def __new__(cls, body):
         ls = []
         for i in body:
             if isinstance(i, CodeBlock):
-                ls += i.body
-            elif isinstance(i, (Assign, For, AugAssign, FunctionalFor,
-                            Application, Expr, IfTernaryOperator, EmptyLine)):
-                ls.append(i)
+                ls += list(i.body)
```

Comment:
I cannot see why one needs to convert `i.body` to a list. In order to extend a list (by either using `+=` or calling the `extend` method) it is sufficient that the argument is an iterable object.

Reply:
@saidctb advised doing this as he wants `i.body` to become a tuple. But if `+=` works with a tuple then I agree that this isn't necessary

Reply:
I didn't know that it works with a tuple , then yes we don't need to convert it

Reply:
```
In [1]: a = list()

In [2]: a.extend?                                                                                                  
Docstring: L.extend(iterable) -> None -- extend list by appending elements from the iterable
Type:      builtin_function_or_method
```